gapis/api/vulkan/transform_command_disabler.go

```diff
@@ -0,0 +1,418 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vulkan
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/transform"
+	"github.com/google/gapid/gapis/memory"
+)
+
+type commandDisabler struct {
+	disabledCommands []api.SubCmdIdx
+	cmdsOffset       uint64
+
+	stateMutator transform.StateMutator
+	allocations  *allocationTracker
+
+	readMemoriesForSubmit []*api.AllocResult
+	readMemoriesForCmd    []*api.AllocResult
+	writeMemoriesForCmd   []*api.AllocResult
+
+	pool VkCommandPool
+}
+
+func newCommandDisabler(ctx context.Context, cmdsOffset uint64) *commandDisabler {
+	return &commandDisabler{
+		disabledCommands:      make([]api.SubCmdIdx, 0, 0),
+		readMemoriesForSubmit: make([]*api.AllocResult, 0),
+		stateMutator:          nil,
+		allocations:           nil,
+		cmdsOffset:            cmdsOffset,
+	}
+}
+
+// Remove removes a draw call command from a command buffer.
+func (disablerTransform *commandDisabler) remove(ctx context.Context, id api.SubCmdIdx) error {
+	if len(disablerTransform.disabledCommands) > 0 {
+		return fmt.Errorf("Multiple Drawcall removal not implemented")
+	}
+
+	if len(id) == 0 {
+		return fmt.Errorf("Requested id is empty")
+	}
+
+	if len(id) > 4 {
+		return fmt.Errorf("Drawcall removal from secondary command buffer is not implemented")
+	}
+
+	id[0] = id[0] + disablerTransform.cmdsOffset
+
+	disablerTransform.disabledCommands = append(disablerTransform.disabledCommands,
+		append(api.SubCmdIdx{}, id...))
+	return nil
+}
+
+func (disablerTransform *commandDisabler) RequiresAccurateState() bool {
+	return false
+}
+
+func (disablerTransform *commandDisabler) RequiresInnerStateMutation() bool {
+	return true
+}
+
+func (disablerTransform *commandDisabler) SetInnerStateMutationFunction(mutator transform.StateMutator) {
+	disablerTransform.stateMutator = mutator
+}
+
+func (disablerTransform *commandDisabler) BeginTransform(ctx context.Context, inputState *api.GlobalState) error {
+	disablerTransform.allocations = NewAllocationTracker(inputState)
+	return nil
+}
+
+func (disablerTransform *commandDisabler) EndTransform(ctx context.Context, inputState *api.GlobalState) ([]api.Cmd, error) {
+	return nil, nil
+}
```

Comment:
Could this be a place where we could emit warnings about disabled commands that were requested by never matched?

E.g. when I tried with `./gapit profile -disabledcmds [9999, 0, 0, 1]` on a small trace, I didn't get any warning about this command ID being never seen.